morpheus/stages/inference/triton_inference_stage.py

```diff
@@ -702,28 +707,51 @@ def __init__(self,
                  force_convert_inputs: bool = False,
                  use_shared_memory: bool = False,
                  needs_logits: bool = None,
-                 inout_mapping: dict[str, str] = None):
+                 inout_mapping: dict[str, str] = None,
+                 input_mapping: dict[str, str] = None,
+                 output_mapping: dict[str, str] = None):
         super().__init__(c)
 
         self._config = c
 
         if needs_logits is None:
             needs_logits = c.mode == PipelineModes.NLP
 
-        # Combine the pipeline mode defaults with any user supplied ones
-        inout_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {})
+        input_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {}).get("inputs", {})
+        output_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {}).get("outputs", {})
+
+        inout_mapping_ = {}
+        inout_mapping_.update(input_mapping_)
+        inout_mapping_.update(output_mapping_)
+
         if inout_mapping is not None:
             inout_mapping_.update(inout_mapping)
+            input_mapping_.update(inout_mapping)
```

Comment:
We should deprecate `inout_mapping` if we have added separate options for input vs output.